manifests/database.pp

```diff
@@ -144,11 +145,8 @@
           require  => Class['postgresql::server'],
         }
 
-        # When every component has its own server, we have to allow those servers to
-        # access the database from the network. Postgresl allows this via the
-        # pg_hba.conf file. As this file only accepts ip addresses, the ip address
-        # of server and web has to be supplied as an parameter.
-        if $zabbix_web_ip != $zabbix_server_ip {
+        # When database not in some server with zabbix server include pg_hba_rule to server
+        if ($database_host_ip != $zabbix_server_ip) or ($zabbix_web_ip != $zabbix_server_ip){
```

Comment:
Yes, I agree with this logic, but on current and old versions of this module has this rule, if I change this logic, current implements will stop working.

For warrant  the retro-compatibility I just added an "or" condition for this existent rule.